### PR TITLE
[e2e] Fix bugs in apiclient and test cases

### PR DESCRIPTION
**Pull request by lanfon72**
## Changes
- **UPDATE** harvester_api
    - update `managers.base.is_support` to support `Version` type
    - refactor `api.cluster_version` to decouple from `SettingManager`
    - add typing for `VirtualMachineSnapshotManager`
    - refactor `models.backups` to compatible with LSP[^1]
    - fix default value bug in `models.volumes`
- **UPDATE** fixture `vm_mgmt_static` to fix CIDR is missing (from worker node)
- **UPDATE** test case `test_update_backup_by_yaml` to fix default annotations is missing (timing problem)
- **UPDATE** `test_z_terraform` to fix timing issue while deleting

[^1]: https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides

### CODE REVIEW

File: apiclient/harvester_api/models/backups.py [hunk 1]
```diff
@@ -48,11 +50,11 @@ def for_existing(cls, delete_volumes=True):
         return cls(False, delete_volumes=delete_volumes)
 
 
-class SnapshotRestoreSpec(RestoreSpec):
+class SnapshotRestoreSpec(_BaseBackup):
     @classmethod
     def for_new(cls, vm_name):
-        return super().for_new(vm_name, None)
+        return cls(True, vm_name)
```
[the remark this answers is not in the document]
Comment (lanfon72) [reply apiclient/harvester_api/models/backups.py:56]: `new_vm` should be `bool`, it looks a typo.

File: apiclient/harvester_api/managers/base.py [hunk 2]
```diff
@@ -28,7 +28,11 @@ class BaseManager:
 
     @classmethod
     def is_support(cls, target_version):
-        return parse_version(target_version) >= parse_version(cls.support_to)
+        cur_version = parse_version(cls.support_to)
+        try:
+            return parse_version(target_version) >= cur_version
+        except TypeError:
+            return target_version >= cur_version
```
[the remark this answers is not in the document]
Comment (lanfon72) [reply apiclient/harvester_api/managers/base.py:35]: expected types of `target_version` would be `Version | str`, `cur_version: Version`.

File: apiclient/harvester_api/models/backups.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lanfon72) [reply apiclient/harvester_api/models/backups.py:56]: Consider that type file will not affect existing code, let's fix it in #1017.

**Review by albinsun — APPROVED** [review]
LGTM, thanks.

**Pull request merged**